File: NL2SQL Agent Frontend Development/src/app/types/llm.ts
export type LLMProvider =
  | 'openai'
  | 'nvidia'
  | 'anthropic'
  | 'openrouter'
  | 'ollama'
  | 'custom';

export interface LLMProviderOption {
  id: LLMProvider;
  label: string;
  description: string;
  default_model: string;
  default_base_url?: string | null;
  api_key_required: boolean;
  protocol: string;
}

export interface LLMProfile {
  id: string;
  display_name: string;
  provider: LLMProvider;
  model: string;
  base_url?: string | null;
  has_api_key: boolean;
  api_key_masked?: string | null;
  updated_at?: string | null;
}

export interface LLMSettings {
  active_profile_id?: string | null;
  fallback_profile_id?: string | null;
  profiles: LLMProfile[];
  providers: LLMProviderOption[];
}

export interface LLMProfileUpsert {
  id?: string | null;
  display_name: string;
  provider: LLMProvider;
  model: string;
  base_url?: string | null;
  api_key?: string | null;
}

export interface LLMRoutingUpdate {
  primary_profile_id: string;
  fallback_profile_id?: string | null;
}

export interface LLMTestResult {
  success: boolean;
  provider: LLMProvider;
  model: string;
  latency_ms: number;
  message: string;
}

File: app/schemas/llm.py
from __future__ import annotations

from enum import Enum

from .base import SchemaModel
from .pydantic_compat import Field


class LLMProvider(str, Enum):
    OPENAI = "openai"
    NVIDIA = "nvidia"
    ANTHROPIC = "anthropic"
    OPENROUTER = "openrouter"
    OLLAMA = "ollama"
    CUSTOM = "custom"


class LLMProviderOption(SchemaModel):
    id: LLMProvider
    label: str
    description: str
    default_model: str
    default_base_url: str | None = None
    api_key_required: bool = True
    protocol: str = "openai"


class LLMProfileUpsert(SchemaModel):
    id: str | None = None
    display_name: str = Field(min_length=1)
    provider: LLMProvider
    model: str = Field(min_length=1)
    api_key: str | None = None
    base_url: str | None = None


class LLMRoutingUpdate(SchemaModel):
    primary_profile_id: str
    fallback_profile_id: str | None = None


class LLMProfileResponse(SchemaModel):
    id: str
    display_name: str
    provider: LLMProvider
    model: str
    base_url: str | None = None
    has_api_key: bool = False
    api_key_masked: str | None = None
    updated_at: str | None = None


class LLMSettingsResponse(SchemaModel):
    active_profile_id: str | None = None
    fallback_profile_id: str | None = None
    profiles: list[LLMProfileResponse] = Field(default_factory=list)
    providers: list[LLMProviderOption] = Field(default_factory=list)


class LLMTestResult(SchemaModel):
    success: bool
    provider: LLMProvider
    model: str
    latency_ms: float = 0.0
    message: str
